akka-distributed-data/src/main/scala/akka/cluster/ddata/Replicator.scala

```diff
@@ -751,6 +806,21 @@ final class Replicator(settings: ReplicatorSettings) extends Actor with ActorLog
   val serializer = SerializationExtension(context.system).serializerFor(classOf[DataEnvelope])
   val maxPruningDisseminationNanos = maxPruningDissemination.toNanos
 
+  val hasDurableKeys = settings.durableKeys.nonEmpty
+  val durable = settings.durableKeys.filterNot(_.endsWith("*"))
+  val durableWildcards = settings.durableKeys.collect { case k if k.endsWith("*") ⇒ k.dropRight(1) }
+  val durableStore: ActorRef =
+    if (hasDurableKeys) {
+      val props = settings.durableStoreProps match {
+        case Right(p) ⇒ p
+        case Left((s, c)) ⇒
+          val clazz = context.system.asInstanceOf[ExtendedActorSystem].dynamicAccess.getClassFor[Actor](s).get
+          Props(clazz, c).withDispatcher(c.getString("use-dispatcher"))
+      }
+      context.actorOf(props.withDeploy(Deploy.local), "durableStore")
```

Comment:
The `MapDBDurableStore` should handle restart, by not use the `init` when restarted. Good catch. Should watch it from here also, so that exception in constructor is detected in Replicator.